File: LeetCode-Java/Maximum Subarray.java
public class Solution {
    public int maxSubArray(int[] A) {
        int max = A[0];
        int record = A[0];
        for(int i = 1;i < A.length;i ++)
        {
            if(record > 0) record += A[i];
            else record = A[i];
            max = max > record ? max : record;
        }
        return max;
    }
}

File: Leetcode-cpp/maximum_subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int preMax = nums[0];
        int globalMax = nums[0];
        for (int i = 1; i < nums.size(); ++i) {
            preMax = preMax > 0 ? (preMax + nums[i]) : nums[i];
            globalMax = max(preMax, globalMax);
        }
        return globalMax;
    }
};